### ISSUE TITLE
Crashes while playing videos

### ISSUE THREAD

**Issue by radiolondra** [matham/ffpyplayer#105]
Using ffpyplayer and kivy.core.video Video module to play videos, on some videos from time to time I have the following error:

```
[ERROR  ] [ffpyplayer  ] [tls @ 145f4b80] Error in the pull function.
[WARNING] [ffpyplayer  ] [mov,mp4,m4a,3gp,3g2,mj2 @ 03692880] Packet corrupt (stream = 0, dts = 3676672)
[WARNING] [ffpyplayer  ] [mov,mp4,m4a,3gp,3g2,mj2 @ 03692880] .
[ERROR  ] [ffpyplayer  ] [mov,mp4,m4a,3gp,3g2,mj2 @ 03692880] stream 0, offset 0x10cc480: partial file
[ERROR  ] [ffpyplayer  ] [tls @ 145f4b80] The specified session has been invalidated for some reason.
[ERROR  ] [ffpyplayer  ] [h264 @ 146d73c0] Invalid NAL unit size (1526 > 246).
[ERROR  ] [ffpyplayer  ] [h264 @ 146d73c0] Error splitting the input into NAL units.
```

This happens mostly when the video is near the end of its duration or when I pause the video for a time greater than the video duration and I start playing again (immediately crashes with the error above), so I'm suspecting something continues to go on in ffpyplayer or in Video kivy module even if the video is paused. Some wrong sync with ffmpeg?
This happens pratically always streaming a video from an https link. Playing local videos everything seems to work fine (at least up now).
Any idea?

**Comment by matham** [comment]
Unfortunately this message is from the ffmpeg internals and not kivy or ffpyplayer. This message indicates that the video stream is corrupted, so something in your network is corrupting the packets. There's nothing we can do about this.

I would try playing the stream with ffplay (the ffmpeg tool) and see if you can reproduce and maybe you can open a issue with ffmpeg then. You can also try playing a different https stream from a different source and see if the issue goes away. The server streaming the video is probably broken.

**Comment by radiolondra** [comment]
Hi @matham,

I want to specify that this kind of error happens with a lot of https servers and different videos, not with a specific server/video.
It's intermittent, if it happens today it could not happen tomorrow anymore. 
FFPlay plays all the guilty videos fine (yesterday, today and perhaps tomorrow too...), no errors. And VLC too. Tests are made while ffpyplayer is raising the error with the same test video.
For sure somewhere ther's something different about how FFPlay handles the videos or the connections and what FFPyPlayer does.
Anyway.... thank you.

**Comment by matham** [comment]
Hmm, have you tried pausing and resuming with ffplay as well?

Which os are you on? And how did you install ffpyplayer. Because if you installed it from a wheel, its ffmpeg veriom may be different than the ffplay version you tried, if you got ffplay separately. On windows, the wheel includes ffplay so you can try the exact version. On Linux, I'd try installing ffpyplayer from source so it uses the system ffmpeg (see install docs) because maybe that doesn't have this issue.

**Comment by radiolondra** [comment]
I don't know the structure of how ffpyplayer interface itself with ffmpeg (I've no time to study the sources), anyway I installed it in PyCharm using `pip install ffpyplayer` (I'm on Windows 10 here, but I wish to try the same in Linux and MacOS too to see if the same issue will happen).
For my ffmpeg player tests I used [this lib](https://github.com/kkroening/ffmpeg-python) and a simple code:

```
....
       process1 = (
            ffmpeg
                .input(sourcePath)
                .output('pipe:', format='rawvideo', pix_fmt='rgb24')
                .run_async(pipe_stdout=True)
        )

        process2 = subprocess.Popen(
            [
                'ffplay',
                '-f', 'rawvideo',
                '-codec:a', 'aac',
                '-pixel_format', 'rgb24',
                '-video_size', '{}x{}'.format(width, height),
                '-i', 'pipe:',
            ],
            stdin=process1.stdout,
        )
        process1.wait()
        process2.wait()
....
```

For VLC tests I used an old TKinter code I used in the past to embed VLC player in TKinter elements. This is impossible to do using kivy, because kivy, unfortunately, doesn't allow to use widgets' HWNDs (and this IMHO is an incredible, unbelievable, basic LACK!)

I should try to install ffpyplayer from source on Windows, getting the needed ffmpeg libs and headers and recompile it. Maybe this could be a long job to do and I'm not sure about the results (and I could find other issues due to some needed parameters to set while building or something like this)
